metrics-exporter-prometheus/src/builder.rs

```diff
@@ -465,11 +478,40 @@ impl PrometheusBuilder {
     pub fn build(mut self) -> Result<(PrometheusRecorder, ExporterFuture), BuildError> {
         #[cfg(feature = "http-listener")]
         let allowed_addresses = self.allowed_addresses.take();
+        let purge_timeout = self.purge_timeout.take();
 
         let exporter_config = self.exporter_config.clone();
         let recorder = self.build_recorder();
         let handle = recorder.handle();
 
+        // use the handle to recorder
+        // #[cfg(not(feature = "push-gateway"))]
+        if let Some(purge_timeout) = purge_timeout {
+            let recorder_handle = handle.clone();
+            let purger = async move {
+                loop {
+                    tokio::time::sleep(purge_timeout).await;
+                    recorder_handle.purge();
+                }
+            };
+
+            if let Ok(handle) = runtime::Handle::try_current() {
+                handle.spawn(purger);
+            } else {
+                let thread_name = "metrics-exporter-prometheus-purger";
+
+                let runtime = runtime::Builder::new_current_thread()
+                    .enable_all()
+                    .build()
+                    .map_err(|e| BuildError::FailedToCreateRuntime(e.to_string()))?;
+
+                thread::Builder::new()
+                    .name(thread_name.to_owned())
+                    .spawn(move || runtime.block_on(purger))
+                    .map_err(|e| BuildError::FailedToCreateRuntime(e.to_string()))?;
+            };
+        }
```

Comment:
No need for the conditional logic here: just spawn the future directly.

(We already document that this method must be called from within a Tokio runtime or else it will panic.)